cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelper.java

```diff
@@ -139,180 +156,208 @@ private boolean throttlingEnabled() {
+      if (currThrottledReplicas != null && currThrottledReplicas.value().trim().equals(WILDCARD_ASTERISK)) {
+        // The existing setup throttles all replica. So, nothing needs to be changed.
+        continue;
+      }
 
-    // Merge new throttled replicas with existing configuration values.
-    Set<String> newThrottledReplicas = new TreeSet<>(replicas);
-    if (currThrottledReplicas != null) {
-      newThrottledReplicas.addAll(Arrays.asList(currThrottledReplicas.split(",")));
+      // Merge new throttled replicas with existing configuration values.
+      Set<String> newThrottledReplicas = new TreeSet<>(replicas);
+      if (currThrottledReplicas != null && !currThrottledReplicas.value().equals("")) {
+        newThrottledReplicas.addAll(Arrays.asList(currThrottledReplicas.value().split(",")));
+      }
+      ops.add(new AlterConfigOp(new ConfigEntry(replicaThrottleConfigKey, String.join(",", newThrottledReplicas)), AlterConfigOp.OpType.SET));
+    }
+    if (!ops.isEmpty()) {
+      changeTopicConfigs(topic, ops);
     }
-    topicConfigs.setProperty(replicaThrottleConfigKey, String.join(",", newThrottledReplicas));
-    changeTopicConfigs(topic, topicConfigs);
   }
 
-  /**
-   * Update topic configuration properties if a topic exist. No-op for non-existent topic
-   *
-   * @param topicName topic name
-   * @param topicConfigs configuration properties to update
-   */
-  private void changeTopicConfigs(String topicName, Properties topicConfigs) {
+  void changeTopicConfigs(String topic, Collection<AlterConfigOp> ops) throws ExecutionException, InterruptedException {
+    ConfigResource cf = new ConfigResource(ConfigResource.Type.TOPIC, topic);
+    Map<ConfigResource, Collection<AlterConfigOp>> configs = Collections.singletonMap(cf, ops);
     try {
-      ExecutorUtils.changeTopicConfig(_adminZkClient, topicName, topicConfigs);
+      _adminClient.incrementalAlterConfigs(configs).all().get();
+      waitForConfigs(cf, ops);
     } catch (Exception e) {
-      if (!_kafkaZkClient.topicExists(topicName)) {
-        LOG.debug("Failed to change configs for topic {} since it does not exist", topicName);
+      if (!topicExists(topic)) {
+        LOG.debug("Failed to change configs for topic {} since it does not exist", topic);
         return;
       }
       throw e;
     }
   }
 
+  void changeBrokerConfigs(int brokerId, Collection<AlterConfigOp> ops) throws ExecutionException, InterruptedException {
+    ConfigResource cf = new ConfigResource(ConfigResource.Type.BROKER, String.valueOf(brokerId));
+    Map<ConfigResource, Collection<AlterConfigOp>> configs = Collections.singletonMap(cf, ops);
+    _adminClient.incrementalAlterConfigs(configs).all().get();
+    waitForConfigs(cf, ops);
+  }
+
+  boolean topicExists(String topic) {
+    try {
+      return _adminClient.listTopics().names().get().contains(topic);
+    } catch (InterruptedException | ExecutionException e) {
+      return false;
```

Comment:
Would this function return `false` if the adminClient request times out? -- i.e. should we handle the following case separately inside the catch block?
```
if (Errors.REQUEST_TIMED_OUT.exception().getClass() == e.getCause().getClass()) { 
... 
}
````